src/main/ruby/truffleruby/core/string.rb

```diff
@@ -606,7 +606,7 @@ def upto(stop, exclusive = false)
     stop = StringValue(stop)
 
     if stop.bytesize == 1 && bytesize == 1 && self.ascii_only? && stop.ascii_only?
-      enc = Primitive.encoding_ensure_compatible(self.encoding, stop.encoding)
+      enc = Primitive.encoding_ensure_compatible_str(self, stop)
```

Comment:
Since we only use strings here this felt more correct.

Reply:
Yes, good catch